### PR TITLE
key: init at 2.6.3

### PR DESCRIPTION
**Pull request by fgaz**
###### Motivation for this change

Using this at uni. Looks established and maintained enough

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/logic/key/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{ stdenv
+, fetchurl
+, unzip
+, jdk
+, ant
+, jre
+, makeWrapper
+, runCommand
+, key
```
Comment (fgaz) [inline pkgs/applications/science/logic/key/default.nix:9]: Not sure if that's a good idea

Comment (roberth) [reply pkgs/applications/science/logic/key/default.nix:9]: The tricky bit is if you'd `pkgs.key.override x` and use it directly without inserting it into `pkgs` by means of an overlay, you'd have to do something like `let key = pkgs.key.override (x // { inherit key; }; in key`.

Instead, you could do

```
let gitRevision = # ...
    key = stdenv.mkDerivation rec {
      # ...
      passthru.tests.check-version = /* ... */;  # syntactically unchanged but uses the local binding
    }
in key
```

Comment (fgaz) [reply pkgs/applications/science/logic/key/default.nix:9]: don't we basically have the same problem with all nixosTests?

Comment (roberth) [reply pkgs/applications/science/logic/key/default.nix:9]: That's an excuse rather than argument, but I'll leave it up to you.

**Review by roberth — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99761` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>key</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99761` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>key</li>
  </ul>
</details>

**Pull request merged**